src/main/java/edu/wpi/grip/core/Socket.java

```diff
@@ -95,13 +99,13 @@ public T getValue() {
      * @param step The step that this socket is part of.
      */
     public void setStep(Step step) {
```

Comment:
Whatever, this doesn't make any sense but I'll do it if it makes you happy. I don't think its a good design decision. Socket should be empty if unset and if you set it by calling `setStep(Step step)` because that will assign the step into the optional value. I don't think it should be possible to redefine the step as empty after you have assigned it.